bundles/http/Citrix/SAP-Ariba/Ariba_v2.js

```diff
@@ -1,399 +1,410 @@
-				'apikey': sync.integrationParameters.apiKey
-			}
-		})
-		let json = await responseData.json()
-		if (responseData.ok && isObject(json)) {
-			if (notEmptyArray(json.comments)) {
-				sync.dataStore.save('rq_comments', flatComments(json.comments)) //saving comments 
-				delete json.comments //deleting comments from json to gain better performance
-			}
-			if (notEmptyArray(json.approvalRequests)) {
-				groups = await parseAndStoreApprovalRequests(sync, json.approvalRequests, json.totalCost, groups)
-				delete json.approvalRequests
-			}
-			if (notEmptyArray(json.lineItems)) { //saving line items
-				for (let j = 0; j < json.lineItems.length; j++) {
-					if (notEmptyArray(json.lineItems[j].accountings)) {
-						sync.dataStore.save('rq_li_accountings', flatLineItemsAccountings(json.lineItems[j].accountings)) //saving li accounting
-						delete json.lineItems[j].accountings
-					}
-				}
-				sync.dataStore.save('rq_lineitems', flatLineItems(json.lineItems)) //saving line items
-				delete json.lineItems
-			}
-			await sync.dataStore.save('requisition_details', flatRequisition(json))
-		}
-	}
-	return groups
+async function storeRequisition(sync, requisitions, groups = new Map()) {
+    console.log("Loading requisition details, count: " + requisitions.length)
+    for (let requisition of requisitions) { // getting requisition details and additional data
+        let responseData = await sync.client.fetch('requisitions/' + requisition + '?realm=' + sync.integrationParameters.realm, {
+            headers: {
+                'apikey': sync.integrationParameters.apiKey
+            }
+        })
+        let json = await responseData.json()
+        if (responseData.ok && isObject(json)) {
+            if (notEmptyArray(json.comments)) {
+                sync.dataStore.save('rq_comments', flatComments(json.comments)) //saving comments
+                delete json.comments //deleting comments from json to gain better performance
+            }
+            if (notEmptyArray(json.approvalRequests)) {
+                await parseAndStoreApprovalRequests(sync, json.approvalRequests, json.totalCost, groups)
+                delete json.approvalRequests
+            }
+            if (notEmptyArray(json.lineItems)) { //saving line items
+                for (let j = 0; j < json.lineItems.length; j++) {
+                    if (notEmptyArray(json.lineItems[j].accountings)) {
+                        sync.dataStore.save('rq_li_accountings', flatLineItemsAccountings(json.lineItems[j].accountings)) //saving li accounting
+                        delete json.lineItems[j].accountings
+                    }
+                }
+                sync.dataStore.save('rq_lineitems', flatLineItems(json.lineItems)) //saving line items
+                delete json.lineItems
+            }
+            sync.dataStore.save('requisition_details', flatRequisition(json))
+        }
+    }
 }
 
 function parseApprovablesGetDocIds(json, documentsLists) {
```

Comment:
what is actually `json` ? it's some array of some items but definitely not an object nor a JSON